tests/unit/change-gate-test.js

```diff
@@ -2,230 +2,348 @@ import EmberObject, { computed } from '@ember/object';
+        return value.split(/\s+/).length;
+      }),
+      letterCount: changeGate('text', function(value) {
+        return value.split('').length;
+      })
+    });
 
-  assert.equal(p1.get('wordCount'), 3);
-  assert.equal(p2.get('wordCount'), 2);
+    var p = Paragraph.create();
 
-  var p1Observer = 0;
-  var p2Observer = 0;
+    var wordCountObserverCount = 0;
+    var letterCountObserverCount = 0;
 
-  p1.addObserver('wordCount', function() {
-    p1Observer++;
-  });
+    p.addObserver('wordCount', function() {
+      wordCountObserverCount++;
+    });
 
-  p2.addObserver('wordCount', function() {
-    p2Observer++;
-  });
+    p.addObserver('letterCount', function(){
+      letterCountObserverCount++;
+    });
 
-  p1.set('text', 'Foo Bar Bar Boo');
-  assert.equal(p1Observer, 1, "the observer fires once when the value is changed on p1");
+    assert.equal(p.get('wordCount'), 2);
+    assert.equal(p.get('letterCount'), 11);
 
-  p2.set('text', "Bar Foo Foo");
-  assert.equal(p2Observer, 1, "the observer fires once when the value is changed on p2");
+    p.set('text', 'Hello  there');
+    assert.equal(p.get('wordCount'), 2);
+    assert.equal(p.get('letterCount'), 12);
+    assert.equal(letterCountObserverCount, 1, "uneffected observer does not fire when another observer is fired");
 
-  p1.set('text', 'Foo Bar Bar Bar Baa');
-  assert.equal(p1Observer, 2, "change to p1 is only recorded on this object, not the other");
-});
+    p.set('text', 'Hello there you');
+    assert.equal(p.get('letterCount'), 15);
+    assert.equal(p.get('wordCount'), 3);
 
-test('multiple changeGate properties on same object', function(assert) {
-  var Paragraph = EmberObject.extend({
-    text: 'Hello there',
-    wordCount: changeGate('text', function(value) {
-      return value.split(/\s+/).length;
-    }),
-    letterCount: changeGate('text', function(value) {
-      return value.split('').length;
-    })
+    assert.equal(letterCountObserverCount, 2, "intended observer fires when effected");
+    assert.equal(wordCountObserverCount, 1, "uneffected observer does not fire when another observer is fired");
   });
 
-  var p = Paragraph.create();
+  test("changeGate filter is bound to instance that it's attached to", function(assert) {
+    assert.expect(1);
 
-  var wordCountObserverCount = 0;
-  var letterCountObserverCount = 0;
+    var instance;
 
-  p.addObserver('wordCount', function() {
-    wordCountObserverCount++;
-  });
+    var Paragraph = EmberObject.extend({
+      text: 'Hello there',
+      wordCount: changeGate('text', function(value) {
+        assert.equal(this, instance);
+        return value.split(/\s+/).length;
+      })
+    });
 
-  p.addObserver('letterCount', function(){
-    letterCountObserverCount++;
+    instance = Paragraph.create();
+    instance.get('wordCount');
   });
 
-  assert.equal(p.get('wordCount'), 2);
-  assert.equal(p.get('letterCount'), 11);
+  module('setting the `sync` state', function() {
```

Comment:
Here I've duped some of the parameter checking tests to make sure they work when setting the observer `sync` state.